src/sentry/discover/dataset_split.py

```diff
@@ -0,0 +1,386 @@
+import logging
+from typing import Any
+
+import sentry_sdk
+from snuba_sdk import AliasedExpression, And, Column, Condition, CurriedFunction, Function, Op, Or
+
+from sentry.api.utils import get_date_range_from_stats_period
+from sentry.constants import ObjectStatus
+from sentry.discover.arithmetic import is_equation, strip_equation
+from sentry.discover.models import DatasetSourcesTypes, DiscoverSavedQuery, DiscoverSavedQueryTypes
+from sentry.exceptions import InvalidParams
+from sentry.models.project import Project
+from sentry.search.events.builder.discover import DiscoverQueryBuilder
+from sentry.search.events.builder.errors import ErrorsQueryBuilder
+from sentry.search.events.types import SnubaParams
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.query_sources import QuerySource
+from sentry.utils import snuba
+from sentry.utils.dates import outside_retention_with_modified_start, parse_timestamp
+
+logger = logging.getLogger("sentry.tasks.split_discover_query_dataset")
+
+
+TRANSACTION_ONLY_FIELDS = [
+    "duration",
+    "transaction_op",
+    "transaction_status",
+    "measurements[lcp]",
+    "measurements[cls]",
+    "measurements[fcp]",
+    "measurements[fid]",
+    "measurements[inp]",
+    "measurements[ttfb]",
+    "measurements[app_start_cold]",
+    "measurements[app_start_warm]",
+    "measurements[frames_total]",
+    "measurements[frames_slow]",
+    "measurements[frames_frozen]",
+    "measurements[frames_slow_rate]",
+    "measurements[frames_frozen_rate]",
+    "measurements[stall_count]",
+    "measurements[stall_total_time]",
+    "measurements[stall_longest_time]",
+    "measurements[stall_percentage]",
+    "measurements[time_to_full_display]",
+    "measurements[time_to_initial_display]",
+    "span_op_breakdowns[ops.browser]",
+    "span_op_breakdowns[ops.http]",
+    "span_op_breakdowns[ops.db]",
+    "span_op_breakdowns[ops.resource]",
+    "span_op_breakdowns[ops.ui]",
+]
+
+ERROR_ONLY_FIELDS = [
+    "location",
+    "exception_stacks.type",
+    "exception_stacks.value",
+    "exception_stacks.mechanism_type",
+    "exception_stacks.mechanism_handled",
+    "received",
+    "exception_main_thread",
+    "exception_frames.abs_path",
+    "exception_frames.colno",
+    "exception_frames.filename",
+    "exception_frames.function",
+    "exception_frames.in_app",
+    "exception_frames.lineno",
+    "exception_frames.module",
+    "exception_frames.package",
+    "exception_frames.stack_level",
+]
+
+
+def save_split_decision_for_query(
```

Comment:
Lets prefix these with `_` and add a comment explaining that these functions should only be used for the split migration so other people in the org don't try to use 'em